### DISCUSSION TITLE
How to get only text from parsed eocument not the markdown code

### DISCUSSION THREAD

**Discussion by robopzet** [thephpleague/commonmark#975] [Q&A] [answered]
I want to get only the text from a document. The nodes of a parsed document contain `Text` nodes, but they expose the original text, including markup. Is there a way to get more basic tokens, for example for a link get a node with the URL and one with the content?

This is how the text is parsed, based on these [docs](https://commonmark.thephpleague.com/2.4/customization/abstract-syntax-tree/).:

```php
$parser = new MarkdownParser(new Environment());
$doc    = $parser->parse($this->getBody());

foreach ($doc->iterator() as $nod) {
    if ($nod instanceof Text) {
        echo 'Node: ' . get_class($nod) . ': ' . $nod->getLiteral() . "\n";
}
```

**Comment by colinodell** [comment, marked as answer]
The `new Environment()` in your code isn't configured with any parsers or extensions, and therefore the engine doesn't know how to parse those tokens, so it assumes everything it sees is plain text.  Try adding the `CommonMarkCoreExtension`:

```php
$environment = new Environment();
$environment->addExtension(new CommonMarkCoreExtension());

$parser = new MarkdownParser($environment);
$doc    = $parser->parse($this->getBody());

foreach ($doc->iterator() as $nod) {
    if ($nod instanceof Text) {
        echo 'Node: ' . get_class($nod) . ': ' . $nod->getLiteral() . "\n";
    }
}
```

**Comment by robopzet** [comment]
@colinodell ok, I see. I thought the default configuration would treat the input as markdown.

**Reply by colinodell** [reply]
Ah, gotcha. It works that way because parsers can only be *added*, not removed, and some people might not want to support things like blockquote or lists in their applications.  (Perhaps using https://commonmark.thephpleague.com/2.4/extensions/inlines-only/#usage)
